android/app/src/main/java/com/microsoft/codepush/react/CodePushNativeModule.java

```diff
@@ -118,7 +118,12 @@ private void setJSBundle(ReactInstanceManager instanceManager, String latestJSBu
 
     private void loadBundle() {
         clearLifecycleEventListener();
-        mCodePush.clearDebugCacheIfNeeded();
+        try {
+            mCodePush.clearDebugCacheIfNeeded(resolveInstanceManager());
```

Comment:
Can we reuse `final ReactInstanceManager instanceManager = resolveInstanceManager();` variable from line 125 (from below) in `clearDebugCacheIfNeeded` to avoid unnecessary using reflection twice and additional try-catch block?

Reply:
Yeah, we can, but we have some cases when `resolveInstanceManager()` has errors. In this case we should use `clearDebugCacheIfNeeded` with `null` argument. I think that adding this logic into next try-catch block is unnecessary as it make it more difficult for understanding. 
Could you please share your thoughts about it?

Reply:
No worries, I believe current solution fits well. The only thing that I'd probably add is a comment with e.g. reference to the issue or some explanation of the intent because it could be not evident why do we need this logic at all.